physionet-django/user/models.py

```diff
@@ -1164,3 +1171,61 @@ class CodeOfConductSignature(models.Model):
 
     class Meta:
         default_permissions = ()
+
+
+class Event(models.Model):
+    """
+    Captures information on events such as datathons, workshops and classes.
+    Used to allow event hosts to assist with credentialing.
+    """
+    title = models.CharField(max_length=64, blank=True, null=True)
+    description = models.TextField(blank=True, null=True)
+    category = models.PositiveSmallIntegerField(choices=EventCategory.choices(), default=EventCategory.COURSE)
+    host = models.ForeignKey(User, on_delete=models.CASCADE)
+    added_datetime = models.DateTimeField(default=timezone.now)
+    start_date = models.DateField(default=timezone.now)
+    end_date = models.DateField(default=timezone.now)
+    slug = models.SlugField(unique=True, default=get_random_string(length=32))
+
+    class Meta:
+        unique_together = ('title', 'host')
+        permissions = [('create_events', 'Can create events'),
+                       ('view_event_menu', 'Can view event menu')]
+
+    def __str__(self):
+        return self.title
+
+    def get_absolute_url(self):
+        return f"/event/{self.slug}/"
+
+    def save(self, *args, **kwargs):
+        """ Add Slug creating/checking to save method. """
+        self.slug_save()
+        super(Event, self).save(*args, **kwargs)
+
+    def slug_save(self):
+        self.slug = get_random_string(length=20)
+
+    def enroll_user(self, user):
+        """
+        Adds a participant to an event.
+        """
+        try:
+            Participants.objects.create(user=user, event=self)
+            group = Group.objects.get(name='Participant')
```

Comment:
We need to make sure that the Participant and Host permission groups are added somewhere (as a fixture perhaps). This code will raise an error if the permission group doesn't exist.

Reply:
Added the permission groups in https://github.com/MIT-LCP/physionet-build/pull/1653/commits/c18400762ef60cc761330dca1ee85b22b7ab8ec7